login tests

jjanickamatraszek opened the pull request.

jjanickamatraszek committed:
add Login Page and Login Tests

ankochergin commented on src/main/java/com/solvd/pages/android/LoginPage.java:
```diff
@@ -0,0 +1,69 @@
+        typePassword(R.TESTDATA.get("password"));
+        clickLoginBtn();
+        return initPage(getDriver(), MainPageBase.class);
+    }
```
please do not hardcode any data in the page objects. For login() method you can create some AuthUtils. For example, there you can have methods like loginWithDefaultUser(), login(username, password)

ankochergin commented on src/main/java/com/solvd/utils/LoginErrorMessages.java:
```diff
@@ -0,0 +1,11 @@
+    public static final String usernameRequired = "Username is required";
+    public static final String passwordRequired = "Password is required";
+    public static final String credsDontMatch = "Username and password do not match any user in this service.";
+}
```
I think it can be enum class `LoginErrorMessage`

ankochergin commented on src/test/resources/testNG/testNG.xml:
```diff
@@ -0,0 +1,9 @@
+<suite name="Regression Suite">
+    <test name="Login">
+        <classes>
+            <class name="com.solvd.login.LoginPageTest"/>
```
please specify each method here using <include>, it helps to support tests later

jjanickamatraszek committed:
update Login Tests

- add AuthUtils
- change LoginErrorMessage to enum
- list login methods in testNG xml

ankochergin commented on src/test/java/com/solvd/login/LoginPageTest.java:
```diff
@@ -0,0 +1,77 @@
+
+    @Test
+    public void loginWithValidCredentialsTest() {
+        AuthUtils authUtils = new AuthUtils();
```
you can create authUtils object in the class or:
1. Create some SauceDemoBaseTest
2. Create there authUtils object
3. Extend all test classes from SauceDemoBaseTest

ankochergin commented on src/main/java/com/solvd/utils/AuthUtils.java:
```diff
@@ -0,0 +1,26 @@
+    public LoginPageBase login(String username, String password) {
+        inputLoginCredentials(username, password);
+        return initPage(getDriver(), LoginPageBase.class);
+    }
```
I believe you can keep only 1 method login(), 2 methods are redundant

ankochergin commented on src/main/java/com/solvd/utils/AuthUtils.java:
```diff
@@ -0,0 +1,26 @@
+    public MainPageBase loginWithDefaultUser() {
+        inputLoginCredentials(R.TESTDATA.get("username"), R.TESTDATA.get("password"));
+        return initPage(getDriver(), MainPageBase.class);
+    }
```
```suggestion

    public MainPageBase login(String username, String password) {
        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
        loginPage.typeName(username);
        loginPage.typePassword(password);
        loginPage.clickLoginBtn();
        return initPage(getDriver(), MainPageBase.class);
    }

    public MainPageBase loginWithDefaultUser() {
        return  login(R.TESTDATA.get("username"), R.TESTDATA.get("password"));
    }
```

jjanickamatraszek committed:
update auth methods, add base test class

jjanickamatraszek committed:
add login page for iOS
